firefox: use nss without p11-kit

mweinelt opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Quickfix to allow firefox to recognize certificates as trusted by
Mozilla.

Downside: NixOS managed certificates will not be passed into firefox anymore until we fixup cacert.

Related: #126065

CC: @rnhmjoj @NixOS/nixos-release-managers 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mweinelt committed:
firefox: use nss without p11-kit

Quickfix to allow firefox to recognize certificates as trusted by
Mozilla.

Related: #126065

dasJ commented:
Does this also need a backport to 20.09?

andir commented:
Don't care about Firefox. Do as you see fit.

jonringer commented on pkgs/applications/networking/browsers/firefox/common.nix:
```diff
@@ -116,7 +116,7 @@ let
                 else stdenv;
 
-  nss_pkg = if lib.versionOlder ffversion "83" then nss_3_53 else nss;
+  nss_pkg = if lib.versionOlder ffversion "83" then nss_3_53 else nss.override { useP11kit = false; };
```
do you mind adding a comment, for future contributors as well as seems unusual

mweinelt replied:
Yes, done.

jonringer reviewed (approved):
LGTM

rnhmjoj commented:
Can you also comment out firefox from `nixos/tests/custom-ca.nix`? Otherwise the test will break.

mweinelt committed:
nixos/tests/custom-ca: disable firefox test integration

Firefox has been decoupled from the system certificate store since the
nss p11-kit integration in combination with our cacert package does not
expose CKA_NSS_MOZILLA_CA_POLICY, which among other things is required
for addon updates.

mweinelt commented:
@GrahamcOfBorg test custom-ca

mweinelt commented:
Tested on x86_64-linux, built on aarch64-linux. The test continues to request screendumps.
